testdata/src/go.uber.org/functioncontracts/inference/functioncontracts-with-inference.go

```diff
@@ -112,14 +107,45 @@ func barNested2() {
 
 // Test the contracted function is called by another function.
 // contract(nonnil -> nonnil)
-func fooParamCalledInAnotherFunction(s *int) *int { // want "^ Annotation on Param 0.*\n.*Must be NILABLE.*\n.*AND.*\n.*Must be NONNIL.*NONNIL$"
-	sink(*s)
-	return new(int)
+func fooParamCalledInAnotherFunction(x *int) *int { // want "^ Annotation on Param 0.*\n.*Must be NILABLE.*\n.*AND.*\n.*Must be NONNIL.*NONNIL$"
+	if x != nil {
+		return new(int)
+	}
+	if rand.Float64() > 0.5 {
+		return new(int)
+	} else {
+		sink(*x) // "nilable value dereferenced" wanted
+		return nil
+	}
 }
 
 func barParamCalledInAnotherFunction() {
-	var s *int
-	call(fooParamCalledInAnotherFunction(s))
+	var x *int
+	call(fooParamCalledInAnotherFunction(x))
 }
 
 func call(x *int) {}
+
+// Test a contracted function is called multiple times in another function.
+// contract(nonnil->nonnil)
+func fooReturnCalledMultipleTimesInTheSameFunction(x *int) *int { // want "^ Annotation on Result 0.*\n.*Must be NILABLE.*\n.*AND.*\n.*Must be NONNIL.*NONNIL$"
```

Comment:
Can we get a version of these two functions where both callsites pass `nilable` and make sure two errors are reported? I am still not sure we are checking both calls within the same caller.

Reply:
Nice point, included.